compiler/optimizer/abstractinterpreter/IDT.cpp

```diff
@@ -0,0 +1,205 @@
+      traceMsg(comp(), "%s\n", header);
+
+   if (candidates <= 0) 
+      {
+      return;
+      }
+
+   // print the IDT nodes in BFS
+   TR::deque<TR::IDTNode*, TR::Region&> idtNodeQueue(comp()->trMemory()->currentStackRegion());
+
+   idtNodeQueue.push_back(getRoot());
+
+   while (!idtNodeQueue.empty())
+      {
+      TR::IDTNode* currentNode = idtNodeQueue.front();
+      idtNodeQueue.pop_front();
+
+      int32_t index = currentNode->getGlobalIndex();
+
+      // skip root node
+      if (index != -1) 
+         {
+         int lineSize = 1 + snprintf(NULL, 0, "#IDT: #%d: #%d inlinable @%d -> bcsz=%d %s target %s, static benefit = %d, benefit = %f, cost = %d, budget = %d, callratio = %f, rootcallratio = %f",
+                                     index,
+                                     currentNode->getParentGloablIndex(),
+                                     currentNode->getByteCodeIndex(),
+                                     currentNode->getByteCodeSize(),
+                                     currentNode->getResolvedMethodSymbol()->signature(comp()->trMemory()),
+                                     currentNode->getName(comp()->trMemory()),
+                                     currentNode->getStaticBenefit(),
+                                     currentNode->getBenefit(),
+                                     currentNode->getCost(),
+                                     currentNode->getBudget(),
+                                     currentNode->getCallRatio(),
+                                     currentNode->getRootCallRatio()
+                                     );
+         char *line = (char*) comp()->trMemory()->allocateStackMemory(lineSize);
+         int lineLen = snprintf(line, lineSize, "#IDT: #%d: #%d inlinable @%d -> bcsz=%d %s target %s, static benefit = %d, benefit = %f, cost = %d, budget = %d, callratio = %f, rootcallratio = %f",
+                                index,
+                                currentNode->getParentGloablIndex(),
+                                currentNode->getByteCodeIndex(),
+                                currentNode->getByteCodeSize(),
+                                currentNode->getResolvedMethodSymbol()->signature(comp()->trMemory()),
+                                currentNode->getName(comp()->trMemory()),
+                                currentNode->getStaticBenefit(),
+                                currentNode->getBenefit(),
+                                currentNode->getCost(),
+                                currentNode->getBudget(),
+                                currentNode->getCallRatio(),
+                                currentNode->getRootCallRatio()
+                                );
+         TR_ASSERT_FATAL(lineLen = lineSize -1, "Truncation in log line");
+         if (verboseInlining){
+            TR_VerboseLog::vlogAcquire();
+            TR_VerboseLog::writeLine(TR_Vlog_BI, "%s\n", line);
+            TR_VerboseLog::vlogRelease();
```

Comment:
This (and the similar code above with `header`) has the same locking behaviour as `writeLineLocked()`. If you mean to use `vlogAcquire()`/`vlogRelease()` to keep the IDT contiguous, then `vlogAcquire()` must happen once just before the first line of the IDT is printed, and `vlogRelease()` just once after the last line

Reply:
Thanks for updating this! I've left an additional comment at the new `vlogAcquire()` call site about some lingering locking issues